src/IO/ReadHelpers.h

```diff
@@ -1100,14 +1100,26 @@ inline void readBinary(LocalDate & x, ReadBuffer & buf) { readPODBinary(x, buf);
 
 
 template <std::endian endian, typename T>
-requires is_arithmetic_v<T> && (sizeof(T) <= 8)
+requires std::is_integral_v<T> && (sizeof(T) <= 8)
 inline void readBinaryEndian(T & x, ReadBuffer & buf)
 {
     readPODBinary(x, buf);
     if constexpr (std::endian::native != endian)
         x = std::byteswap(x);
 }
 
+template <std::endian endian, typename T>
+requires is_decimal<T> || std::is_floating_point_v<T>
+inline void readBinaryEndian(T & x, ReadBuffer & buf)
+{
+    readPODBinary(x, buf);
+    if constexpr (std::endian::native != endian)
+    {
+        char *start = reinterpret_cast<char *>(&x), *end = start + sizeof(T);
+        std::reverse(start, end);
+    }
+}
+
```

Comment:
I think you can get rid of all these changes and its newest version that uses `transformEndianness` should take care of things.

Reply:
Done, used your latest code, reverted this change.